src/WebJobs.Extensions.DurableTask/EntityScheduler/TypedInvocationExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Extends the durable entity context to support reflection-based invocation of entity operations.
+    /// </summary>
+    public static class TypedInvocationExtensions
+    {
+        /// <summary>
+        /// Dynamically dispatches the incoming entity operation using reflection.
+        /// </summary>
+        /// <typeparam name="T">The class to use for entity instances.</typeparam>
+        /// <returns>A task that completes when the dispatched operation has finished.</returns>
+        /// <exception cref="AmbiguousMatchException">If there is more than one method with the given operation name.</exception>
+        /// <exception cref="MissingMethodException">If there is no method with the given operation name.</exception>
+        /// <exception cref="InvalidCastException">If the method has more than one argument.</exception>
```

Comment:
`InvalidCastException` has a pretty well-known meaning, and I feel like this use of it is confusing. How about `InvalidOperationException`? That one coincidentally has a double meaning in the sense that this is an invalid entity operation definition.

Reply:
yeah, I could not find one that I was completely happy with but `InvalidOperationException` is fine.